### ISSUE TITLE
Items listed in subscription email limited to 10 per collection/community

### ISSUE THREAD

**Issue by uofmsean** [DSpace/DSpace#9000]
**Describe the bug**
DSpace 7.5 - Subscription emails for modified items only contain at most 10 items per collection or community.

It seems to be caused by:
1. [maxResults in DiscoveryQuery is defaulted to -1](https://github.com/DSpace/DSpace/blob/fadfdc6ed20632ddc093ad125c1a2c6334171631/dspace-api/src/main/java/org/dspace/discovery/DiscoverQuery.java#L41)
2. [SolrQuery row limit in resolveToSolrQuery() is set only if maxResults != -1](https://github.com/DSpace/DSpace/blob/fadfdc6ed20632ddc093ad125c1a2c6334171631/dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java#L894-L896)
3. Therefore, the default row limit of 10 for a SolrQuery is used

**To Reproduce**
Steps to reproduce the behaviour:
1. Update more than 10 items in one collection
5. Subscribe to that collection
6. Run "dspace subscription-send -f D" the next day
7. Only 10 items are listed in the subscription email for the collection

**Expected behavior**
All updated (or only new) items should be listed and not limited to 10. It would be nice to have a new parameter to specify the limit of items that will be listed per collection and/or community.

**Related work**
#8880

**Comment by tdonohue** [comment]
Agreed this sounds like a bug.  It likely should _not_ be unlimited though (e.g. a batch upload of thousands of items shouldn't result in thousands displayed in this email).  So it may make sense to make it configurable in some way.

Needs a volunteer to investigate options. If it's minimal changes / backwards compatible with 7.6, this could potentially be included in a 7.6.x release.
